### DISCUSSION TITLE
Section 't5uid1' is not a valid config, after update

### DISCUSSION THREAD

**Discussion by datendieman** [Desuuuu/DGUS-reloaded-Klipper#26] [Q&A]
Hey all,

Just did a klipper update and now it doesnt seem to recognize the "t4uid1" parameter in the config.

Any ideas how to fix this?

Thank you!

**Comment by Desuuuu** [comment]
This essentially means you are not running my fork anymore.

**Reply by datendieman** [reply]
> This essentially means you are not running my fork anymore.

thanks for the response! for future reference, what is the proper way of updating klipper if I'm running this fork?
I pushed the update button in fluid and it seems to have overwritten the original install.

**Reply by Desuuuu** [reply]
I'm guessing you could configure Moonraker's updater to use the proper repo. I can't help much more since I'm not using this particular setup though.

**Reply by Newcal** [reply]
I just reflash the printer with klipper and the modified firmware with your fork and still having the same message

**Comment by hieppo** [comment]
Instructions are a bit confusing.
I have installed your dgus-display klipper branch and it is running currently. I have build the firmware and the log shows this:
Version: dgus-display-211225-6-g90ebed1f
  Preprocessing out/src/generic/armcm_link.ld
  Linking out/klipper.elf
  Creating hex file out/klipper.bin

I have flashed that to my CR-200B (it uses the same DWIN screen as the CR10S Pro and others)

I tried adding to the Klipper printer.cfg
[t5uid1]
firmware: dgus_reloaded
Klipper firmware would not start complaining section does not recognised.

I then tried:
[dgus_display]
type: t5uid1
uart_bus: usart3
implementation: dgus_printer_menu
volume: 50

This does allow me to boot up klipper and the DWIN screen does go beyond the bootup log (but not Klipper log) and into strange Chinese language and unresponsive.

I did flash the DWIN_SET. But the the final summary shows no update to any BMP. This may be why the screen is not showing properly? Any idea why it did not flash?

I might just found the answer. The CR-200B LCD is a DGUS2 display. Lots of the file names are different and the CFG file is not a T5UID1 but a T5LCFG. There probably some similarity but it is different enough that it will not work straight out without some reworking.

**Reply by ratomas108** [reply]
Hello,
I have T5L1 display ( Sermoon D1 ). I have to compiled source code in SW ( DGUS_Tool_V8.2.1.14 ). Now it is working. I used original T5L.cfg file, generated new bin files, recompiled icons, buttons, screens.

**Reply by hieppo** [reply]
Hi could you please provide some steps you did to get the display to work

**Comment by ratomas108** [comment]
Here are some steps: https://github.com/Desuuuu/klipper/discussions/122#discussioncomment-4147890. I will write some steps later.

**Reply by ratomas108** [reply]
1. Download stock FW, source project from Desuuuu.
2.  Open stock FW and copy *.HZK and *.CFG files.
3. Open DWprj.hmi file in DGUS Tool.
4. Load *.CFG ( Setting\DGUS - CFG Edit - Open CFG )
5. Generate bin files - File\Generate
![image](https://user-images.githubusercontent.com/106281051/202204723-2091a266-d4c8-46f0-83c5-59ef293c998c.png)
6. Generate ICL files for 24_icons, 27_buttons, 30_progress_left, 37_progress_right, screens
Setting\DGUS - ICL Tool - Genereate ICL
![image](https://user-images.githubusercontent.com/106281051/202208218-580a9dc6-56d6-477d-b4b2-448ff32ff4d0.png)
7. Create your DWIN_SET folder with files:
3 bin files, 5 icon files ( generated from DGUS Tool ), 1 HZK, 1 CFG file
![image](https://user-images.githubusercontent.com/106281051/202206965-353412b6-7d44-460d-816f-a97ef58799c4.png)
8. Flash FW

**Comment by ratomas108** [comment]
Files are in your project
...\DGUS-reloaded-Klipper-master\project\DWIN_SET. Check date of *.bin
files.

**Reply by hieppo** [reply]
Found it afterwards so I deleted the message. Anyway, my new DWIN_SET folder will only contain the 10 files you have in step 7? What about all those bmp, wav and ico files that normally seen in a DWIN_SET flash?

**Comment by ratomas108** [comment]
From BMP files you will create  5 lcon library files. WAV Is not neceseary.

**Reply by hieppo** [reply]
Hi,
I flashed those 10 files and then flashed desuuuu klipper firmware version. I tried to enable it in printer.cfg with:
[dgus_display]
type: t5uid1
uart_bus: usart3
implementation: dgus_printer_menu
volume: 30

That was the only way to bring up the display. However, I see the OEM Creality images and the touches are not properly assigned. The menu does not really work as intended.

Any  assistance would be appreciated.

**Reply by ratomas108** [reply]
When you flash FW succesfully, you should see Klipper logo. Try to rename icon library file for screens to 000_boot. I think you can flash only files from list during flashing.
![image](https://user-images.githubusercontent.com/106281051/202762569-f1c2ff06-a8d4-4ace-abc6-7fd0c90d6ec9.png)

**Reply by hieppo** [reply]
when I flash, I see 1 CFG, 3 BIN, 1 HZK and 4 ICL files downloaded. I rename 000_boot.icl but it did not pick it up
BTW, based on your steps, I should use the CFG from my OEM DWIN_SET. For the HZK file, I should use the one from desuuuu project/DWIN_SET?

**Reply by ratomas108** [reply]
When I was flashing display, I made few attempts. Repeat flashing. Try to generate 000_boot.
![image](https://user-images.githubusercontent.com/106281051/202765957-309588f9-68c8-4b47-9f0c-9fc4e1467419.png)
![image](https://user-images.githubusercontent.com/106281051/202766086-45e11658-204d-4d70-99a2-091eb2df3bdf.png)
![image](https://user-images.githubusercontent.com/106281051/202767491-dc7d5a2b-c23b-48b9-b4ed-d904382f770a.png)

**Reply by hieppo** [reply]
I am using the dgus-display klipper branch. So the t5uid1 section is not used. It is replaced with dgus_display. Again, not only the klipper boot image is not displayed at boot time but the menu is not working. I guess there is more required to work with my Creality CR200B

**Reply by ratomas108** [reply]
Try to copy all bmp files for screens to DWIN_SET folder and flash again.

**Reply by hieppo** [reply]
the only thing I am able to do with this desuuuu version is to display the files to print menu of Creality. But it is using the Chinese language and the touch zone is incorrectly position. Before this, all I see is the OEM Creality boot screen and it stays there.  It is one step up but it goes nowhere.

**Reply by ratomas108** [reply]
Your config is:
[dgus_display]
type: t5uid1
uart_bus: usart3
implementation: dgus_printer_menu
volume: 50
Are you trying dgus printer menu or dgus reloaded klipper for display?

**Reply by hieppo** [reply]
I am using this desuuuu klipper branch call 
dgus-display. klipper
dgus-display: dgus-display-211225-360-g25110253 

Based on your screenshot, you are using the dgus-reloaded-2112xxxxx which is the older branch I believe and it uses a different section for the printer.cfg.

**Reply by hieppo** [reply]
i think the t5l1 flash did not change anything since it is not displaying anything that looks like klipper. I still see Crealiity logo. The flash indicate downloaded but does not overwrite default display.

**Reply by ratomas108** [reply]
Create Dwin_set folder with 10 files + all BMP files. Flash again.

**Reply by hieppo** [reply]
i tried that previously and it still does not change the screen. I can tell right away when the printer boots up, I still see the Creality bootscreen.

**Reply by hieppo** [reply]
You've indicate to create 5 icon files where the last one is 000_screens.icl. What image do I select for this? BTW, there are icl files already created with custom Klipper images? Why can't we use them?

**Reply by ratomas108** [reply]
24_icons - generate from 5 files
27_buttons - generate from 19 files
30_progress_left - generate from 101 files
37_progress_right - generate from 101 files
000_boot - generate from 37 files

Stock FW for my display:
![image](https://user-images.githubusercontent.com/106281051/202872796-3c461385-d0c7-47de-ada7-068696627bd9.png)

You should create similar files.

**Reply by hieppo** [reply]
Based on your instructions:
1. I use the CFG from Creality OEM which is T5LCFG.CFG. I copy this file to the *new* DWIN_SET folder and open it in DGUS tool.
2. I use the HZK file from Creality OEM or Desuuuu stock project? I copy this to the *new* DWIN_SET folder
3. I open the DWprj.hmi from Desuuuu stock project folder which lists out all the images to the left.
4. File/Generate which will generate the 3 bin files (stored in the project/DWIN_SET folder). I copy it out of there into the *new* DWIN_SET folder
5. I use ICL Generate 24_icons.icl from the 24_icons folder (5 bmp) under Desuuuu project and copy it to the *new* DWIN_SET folder
6. I use ICL Generate 27_buttons.icl from the 27_buttons folder (19 bmp) under Desuuuu project and copy it to the *new* DWIN_SET folder
7. I use ICL Generate 30_progress_left.icl from the 30_progress_left (101 bmp) folder under Desuuuu project and copy it to the *new* DWIN_SET folder
8. I use ICL Generate 37_progress_right.icl from the 37_progress_right (101 bmp) folder under Desuuuu project and copy it to the *new* DWIN_SET folder

9. What folder that contains the 37 bmp files do I create 000_boot.icl?
10. What similar files should I create with my stock FW?

The last 2 steps are unknown. The farthest I was able to see after the flash is a black screen instead of the Creality boot screen. Once booted, my screen is black with 4 fuzzy squares. It is definitely different than OEM but it is not definitely functional.

**Reply by ratomas108** [reply]
Ad 2. Try both.
Ad 9. Generate 000_boot.icl from DWIN_SET folder ( 000_boot ... 250_kill ).
Ad 10. Compare stock FW and your new FW. Structure must be similar.

**Reply by hieppo** [reply]
When you say to compare with stock FW, what exactly am I comparing? Is There are more *.icl files and they have different names? There is one common ##_control.icl. There are bmp files that have different numbering scheme and names. So I am not sure what to compare to Desuuuu DWIN_SET.

**Comment by hieppo** [comment]
Hey desuuuu,
Can you provide some starting point on what you used to develop the DWIN_SET file? We have the DGUS tool, we should be able to customize for our specific display based on what ratomas108 was able to achieve to get it to work with his Creality printer.

**Reply by ratomas108** [reply]
Some information:
https://github.com/Desuuuu/klipper/discussions/122#discussioncomment-4578285
